### PR TITLE
docs: private-network-tutorial

### PR DESCRIPTION
**Pull request by JustinCast**
### GH Issue

#268

### Steps to test
1. `yarn start`
1. Go to localhost:3000 and search instalacion-de-red-privada

#### CheckList
- [x] Follow proper Markdown format
- [x] The content is adequate
- [x] The content is available in both english and spanish
- [x] I Ran a spell check

### CODE REVIEW

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+- [Nodo semilla: configuración y ejecución](#nodo-semilla-configuración-y-ejecución)
+  - [`config.ini`](#configini-2)
+  - [`genesis.json`](#genesisjson-2)
+  - [`start.sh`](#startsh-2)
+- [Detener el servicio nodeos](#detener-el-servicio-nodeos)
+- [Reiniciar el servicio nodeos](#reiniciar-el-servicio-nodeos)
+- [Reiniciar nodeos desde cero](#reiniciar-nodeos-desde-cero)
+- [Interactuar con la red](#interactuar-con-la-red)
+  - [Obtener datos almacenados](#obtener-datos-almacenados)
+# Instalación y configuración de una red privada
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:37]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+Se pueden diseñar varias topologías en el sentido de cantidad de nodos o redundancia de los datos, sin embargo, para la practicidad de este tutorial, se utiliza la siguiente topología como referencia principal:
+<p style={{ align: "center" }}>
+  <img src={useBaseUrl( '/img/private-network-installation-tutorial/initial-topology.png' )} width="100%" />
+</p>
+
+# Requisitos previos de software
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:43]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 3]
```diff
@@ -0,0 +1,778 @@
...
+A la hora de configurar una red privada EOSIO hay algunos requisitos de instalación de software:
+1. Instalación de los binarios precompilados de EOSIO
+1. Instalación de los binarios EOSIO.CDT
+
+## Instalación de los binarios precompilados de EOSIO
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:48]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 4]
```diff
@@ -0,0 +1,778 @@
...
+Ejecute los siguientes comandos para instalar los binarios precompilados de EOSIO:
+```bash
+$ wget https://github.com/eosio/eos/releases/download/v2.0.9/eosio_2.0.9-1-ubuntu-18.04_amd64.deb
+$ sudo apt install ./eosio_2.0.9-1-ubuntu-18.04_amd64.deb
+```
+Una vez instalados los binarios de EOSIO; nodeos, una configuración de blockchain desde cero e interacciones con redes P2P son algunas de las funcionalidades disponibles que ofrece el protocolo.
+> Para desinstalar EOSIO ejecute `$ sudo apt remove eosio`
+## Instalación de los binarios EOSIO.CDT (Contract Development Toolkit)
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:56]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 5]
```diff
@@ -0,0 +1,778 @@
...
+EOSIO.CDT es un kit de herramientas que facilita el desarrollo de contratos inteligentes en una red de blockchain EOSIO. Para instalar EOSIO.CDT v1.6.3 ejecute los siguientes comandos:
+```bash
+$ wget https://github.com/eosio/eosio.cdt/releases/download/v1.6.3/eosio.cdt_1.6.3-1-ubuntu-18.04_amd64.deb
+$ sudo apt install ./eosio.cdt_1.6.3-1-ubuntu-18.04_amd64.deb
+```
+# Configuración del nodo génesis
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:62]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 6]
```diff
@@ -0,0 +1,778 @@
...
+    "max_inline_action_size": 4096,
+    "max_inline_action_depth": 4,
+    "max_authority_depth": 6
+  },
+  "initial_chain_id": "0000000000000000000000000000000000000000000000000000000000000000"
+}
+```
+> Esto puede ser cambiado más adelante utilizando la función privilegiada `set_param`. Puedes ver más [aquí](https://guias.eoscostarica.io/docs/aprender-eosio/funciones-importantes#docsNav)
+
+# Iniciar el servicio nodeos
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:101]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 7]
```diff
@@ -0,0 +1,778 @@
...
+    $ chmod 755 genesis_start.sh
+    $ ./genesis_start.sh
+    ```
+    Una vez ejecutado el comando anterior, el nodo génesis que llevará **eosio** como nombre podrá:
+1. Producir bloques
+1. Escuchar peticiones HTTP en `127.0.0.1:8888`.
+1. Escuchar solicitudes de conexión con otros nodos en `127.0.0.1:9010`.
+1. Imprimir la salida de los contratos inteligentes en el terminal
+
+# Registros del servicio nodeos
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:156]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 8]
```diff
@@ -0,0 +1,778 @@
...
+Puedes inspeccionar los registros de servicios de nodeos con los siguientes comandos:
+```bash
+$ cd ~/biosboot/genesis/
+$ tail -f ./blockchain/nodeos.log
+``` 
+> Para salir del modo de impresión de registros, pulse CTRL+C.
+
+# Configuración y compilación de `eosio.contracts`
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:164]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 9]
```diff
@@ -0,0 +1,778 @@
...
+1. Clone y compile `eosio.contracts` `v1.8.0`:
+    ```bash
+    $ cd ~
+    $ git clone https://github.com/EOSIO/eosio.contracts.git eosio.contracts-1.8.x
+    $ cd ./eosio.contracts-1.8.x/
+    $ git checkout release/1.8.x
+    $ ./build.sh
+    $ cd ./build/contracts/
+    ```
+# Despliegue del contrato `eosio.bios`
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:189]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 10]
```diff
@@ -0,0 +1,778 @@
...
+## Activar `PREACTIVATE_FEATURE`
+Debido a las actualizaciones introducidas en la v1.8 y v2.0, es necesario activar una característica del protocolo llamada PREACTIVATE_FEATURE. Para activar esta característica, ejecute el siguiente comando:
+```bash
+$ curl --request POST \
+	--url http://127.0.0.1:8888/v1/producer/schedule_protocol_feature_activations \
+	-d '{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}' 
+```
+## Despliegue de la versión **antigua** de los contratos
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:197]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 11]
```diff
@@ -0,0 +1,778 @@
...
+# RAM_RESTRICTIONS
+cleos push action eosio activate '["4e7bf348da00a945489b2a681749eb56f5de00b900014e137ddae39f48f69d67"]' -p eosio
+
+# WEBAUTHN_KEY
+cleos push action eosio activate '["4fca8bd82bbd181e714e283f83e1b45d95ca5af40fb89ad3977b653c448f78c2"]' -p eosio
+
+# WTMSIG_BLOCK_SIGNATURES
+cleos push action eosio activate '["299dcb6af692324b899b39f16d5a530a33062804e41f09dc97e9f156b4476707"]' -p eosio
+```
+## Desplegar la última versión de los contratos
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:249]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 12]
```diff
@@ -0,0 +1,778 @@
...
+**Salida esperada de consola:**
+```bash
+Reading WAST/WASM from ·/users/documents/eos/contracts/eosio.bios/eosio.bios.wasm...
+Using already assembled WASM...
+Publishing contract...
+executed transaction: 17fa4e06ed0b2f52cadae2cd61dee8fb3d89d3e46d5b133333816a04d23ba991  8024 bytes  974 us
+#         eosio <= eosio::setcode               {"account":"eosio.bios","vmtype":0,"vmversion":0,"code":"0061736d01000000017f1560037f7e7f0060057f7e...
+#         eosio <= eosio::setabi                {"account":"eosio.bios","abi":{"types":[],"structs":[{"name":"transfer","base":"","fields":[{"name"...
+```
+# Nodos productores de bloques: configuración y ejecución
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:263]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 13]
```diff
@@ -0,0 +1,778 @@
...
+Para ejecutar cada nodo es necesario entrar en el directorio, asignar permisos y ejecutar cada archivo `start.sh`:
+```bash
+$ chmod 755 start.sh # asigna permisos de ejecución
+$ ./start.sh
+```
+Una vez realizados los pasos necesarios para poner en marcha los nodos productores, es el momento de configurar las cuentas produder1, producer2 y producer3 como productores en bloque en el **schedule**, para ello es necesario ejecutar el siguiente comando (recuerde sustituir EOS_PUB_DEV_KEY por su respectivo valor):
+```bash
+$ cleos -u http://localhost:8888 push action eosio setprods {"schedule":[{"producer_name":"producer1","authority": [{"threshold":1,"keys":[{"key":"EOS_PUB_DEV_KEY","weight":1}]}]},{"producer_name":"producer2","authority": [{"threshold":1,"keys":[{"key":"EOS_PUB_DEV_KEY","weight":1}]}]},{"producer_name":"producer3","authority": [{"threshold":1,"keys":[{"key":"EOS_PUB_DEV_KEY","weight":1}]}]}]}
+```
+# Nodo API: configuración y ejecución
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:404]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 14]
```diff
@@ -0,0 +1,778 @@
...
+do
+  tail -f /dev/null & wait ${!}
+done
+```
+Para iniciar el nodo, asigne permisos de ejecución a `start.sh` y ejecútelo:
+```bash
+$ chmod 755 start.sh
+$ ./start.sh
+```
+# Nodo semilla: configuración y ejecución
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:537]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 15]
```diff
@@ -0,0 +1,778 @@
...
+do
+  tail -f /dev/null & wait ${!}
+done
+```
+Para iniciar el nodo semilla, simplemente asigne permisos de ejecución al archivo `start.sh` y ejecútelo:
+```bash
+$ chmod 755 start.sh
+$ ./start.sh
+```
+# Detener el servicio nodeos
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:657]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 16]
```diff
@@ -0,0 +1,778 @@
...
+    done
+    echo -ne "\rNode Stopped. \n"
+fi
+```
+Ahora puede ejecutar el archivo `stop.sh`:
+```bash
+$ chmod 755 stop.sh 
+$ ./stop.sh
+```
+# Reiniciar el servicio nodeos
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:682]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 17]
```diff
@@ -0,0 +1,778 @@
...
+--p2p-peer-address localhost:9011 \
+--p2p-peer-address localhost:9012 \
+--p2p-peer-address localhost:9013 \
+--hard-replay-blockchain \
+>> $DATADIR"/nodeos.log" 2>&1 & \
+echo $! > $DATADIR"/eosd.pid"
+```
+> **"perhaps we need to replay"**: este error puede producirse al reiniciar nodeos debido a que falta la bandera `--hard-replay-blockchain` (parámetro que reproduce todas las transacciones del nodo génesis). Para solucionar este error, ejecute el script `hard_replay.sh`.
+
+# Reiniciar nodeos desde cero
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:762]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 18]
```diff
@@ -0,0 +1,778 @@
...
+ls -al
+```
+Si desea eliminar la configuración actual, los datos de la cadena de bloques y los archivos de registro, ejecute los siguientes comandos en el orden en que aparecen:
+```bash
+$ cd ~/biosboot/genesis/
+$ ./stop.sh
+$ ./clean.sh
+$ ./genesis_start.sh
+```
+# Interactuar con la red
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:776]: Title Case

File: docs/tutoriales/instalacion-de-red-privada.md [hunk 19]
```diff
@@ -0,0 +1,778 @@
...
+En este punto, la red es capaz de albergar cualquier contrato. Supongamos que quieres desplegar un contrato como [easycontract](https://github.com/eoscostarica/eoscr-smart-contracts/tree/master/easycontract) el repo de ejemplo de EOS Costa Rica. Después de seguir las instrucciones especificadas allí, puede llamar a la acción `save` de easycontract de la siguiente manera:
+```bash
+cleos -u URL_NODO_AQUI push action easycontract save '{"date":"2021/01/10 21:01:35","filename":"ejemplo.pdf","path":"/var/www/docs"}' -p easycontract@active
+```
+Tras la ejecución, obtendrá una salida en el terminal.
+## Obtener datos almacenados
```
Comment (leisterfrancisco) [inline docs/tutoriales/instalacion-de-red-privada.md:782]: Title Case

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Review by leisterfrancisco — DISMISSED** [review]
Good work 👍

**Comment by Edgar097** [comment]
@JustinCast Shouldn't it be added to the sidebar? I think we are using H2 titles and we leave H1 to the section title.

**Review by xavier506 — APPROVED** [review]
LGTM

**Review by xavier506 — CHANGES_REQUESTED** [review]
@JustinCast 

Please have a look at the header tags, we should use up to h2 tags (not h1)

**Comment by JustinCast** [comment]
Gonna check now

**Comment by JustinCast** [comment]
> @JustinCast Shouldn't it be added to the sidebar? I think we are using H2 titles and we leave H1 to the section title.

Lemme know if there's anything wrong. It's now fixed.

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**